prow/repoowners/repoowners.go

```diff
@@ -215,22 +218,44 @@ func (c *Client) LoadRepoOwners(org, repo, base string) (RepoOwner, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	entry, ok := c.cache[fullName]
-	if !ok || entry.sha != sha || entry.owners == nil || entry.owners.enableMDYAML != mdYaml {
+
+	syncEntry := func(doDiff bool) error {
 		gitRepo, err := c.git.Clone(cloneRef)
 		if err != nil {
-			return nil, fmt.Errorf("failed to clone %s: %v", cloneRef, err)
+			return fmt.Errorf("failed to clone %s: %v", cloneRef, err)
 		}
 		defer gitRepo.Clean()
+
+		if doDiff {
+			changes, err := gitRepo.Diff(sha, entry.sha)
+			if err != nil {
+				return fmt.Errorf("failed to diff %s with %s", sha, entry.sha)
+			}
+			if func() bool {
```

Comment:
Please add some comments to explain what this code is doing. In particular this anonymous function.

Reply:
added

Reply:
Not sure the comment helps -- why is this in a func? It has no args, is not persisted and is just called once? Maybe prefer:


```go
shouldReload := true
for _, change := range changes {
	if mdYaml && strings.HasSuffix(change, ".md") ||
		strings.HasSuffix(change, aliasesFileName) ||
		strings.HasSuffix(change, ownersFileName) {
		shouldReload = false
	}
}

if shouldReload {
    entry.sha = sha
    return nil
}
```

Reply:
fix